apps/wing/src/commands/pack.ts

```diff
@@ -0,0 +1,83 @@
+import * as path from "path";
+import * as fs from "fs";
+import * as cp from "child_process";
+
+export interface PackageOptions {
+  /**
+   * Directory to to save the generated package to.
+   */
+  readonly outdir: string;
+}
+
+export async function pack(options: PackageOptions): Promise<string> {
+  // check npm is installed
+  try {
+    cp.execSync("npm --version", { stdio: "ignore" });
+  } catch {
+    throw new Error(`npm is not installed. Install it from https://www.npmjs.com/get-npm`);
+  }
+
+  // check package.json exists
+  const currentDir = process.cwd();
+  const pkgJsonPath = path.join(currentDir, "package.json");
+  if (!fs.existsSync(pkgJsonPath)) {
+    throw new Error(`No package.json found in the current directory. Run \`npm init\` first.`);
+  }
+
+  const pkgJson = JSON.parse(fs.readFileSync(pkgJsonPath, "utf8"));
+  pkgJson.wing = pkgJson.wing ?? {};
+
+  // check entrypoint file is valid and is specified in package.json
+  const entrypoint = pkgJson.wing.entrypoint ?? "lib.w";
+  if (path.isAbsolute(entrypoint)) {
+    throw new Error(
+      `Entrypoint file "${entrypoint}" must be a relative path. The entrypoint is specified in package.json:\n\n  "wing": {\n    "entrypoint": "lib.w"\n  }\n`
+    );
+  }
+  if (!fs.existsSync(entrypoint)) {
+    throw new Error(
+      `Entrypoint file "${entrypoint}" does not exist. The entrypoint is specified in package.json:\n\n  "wing": {\n    "entrypoint": "lib.w"\n  }\n`
+    );
+  }
+  pkgJson.wing.entrypoint = entrypoint;
+
+  // check source files will be included in tarball
+  const pkgJsonFiles = pkgJson.files ?? [];
+  const expectedGlobs = ["**/*.js", "**/*.w", "!/target/**"];
+  for (const glob of expectedGlobs) {
+    if (!pkgJsonFiles.includes(glob)) {
+      pkgJsonFiles.push(glob);
+    }
+  }
+  pkgJson.files = pkgJsonFiles;
+
+  // check package.json has required fields
+  const requiredFields = ["name", "version", "description", "author", "license"];
+  for (const field of requiredFields) {
+    if (pkgJson[field] === undefined) {
+      throw new Error(`Missing required field "${field}" in package.json`);
+    }
+  }
+
+  // check if "main" points to a valid file, and if not, create a dummy file
+  let main = pkgJson.main ?? "index.js";
+  if (!fs.existsSync(main)) {
+    const lines = [];
+    lines.push("// This file was generated by wing pack");
+    lines.push(`console.log("${pkgJson.name} isn't ready for JS consumption yet!")`);
```

Comment:
```suggestion
    lines.push(`console.log("${pkgJson.name} is a winglang library and cannot be used from javascript (yet)")`);
```

Reply:
@eladb Any reason not to just say "wing" here? Seems like an unambiguous situation

```suggestion
    lines.push(`console.log("${pkgJson.name} is a wing library and cannot be used from javascript (yet)")`);
```

Reply:
If I see this error I likely don't know what wing is... saying winglang will prompt me to look it up and find a whole new world of goodness joy love and creativity.